spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/util/ReflectionUtil.scala

```diff
@@ -99,5 +106,20 @@ object ReflectionUtil {
     setters(implicitly[TypeTag[T]].tpe)
   }
 
+  /** Creates a corresponding `TypeTag` for the given `Type`.
+    * Allows to use reflection-created objects in APIs expecting `TypeTag`. */
+  def typeToTypeTag[T](tpe: Type): TypeTag[T] = {
```

Comment:
Oh yes, good catch. I don't know how many things are computed lazily underneath and are not thread-safe in 2.10, so it won't hurt to enclose it all in synchronized.